src/Base/Reader.h

```diff
@@ -217,13 +217,23 @@ class BaseExport XMLReader: public XERCES_CPP_NAMESPACE_QUALIFIER DefaultHandler
     unsigned int getAttributeCount() const;
     /// check if the read element has a special attribute
     bool hasAttribute(const char* AttrName) const;
-    /// return the named attribute as an integer (does type checking)
-    long getAttributeAsInteger(const char* AttrName) const;
-    unsigned long getAttributeAsUnsigned(const char* AttrName) const;
-    /// return the named attribute as a double floating point (does type checking)
-    double getAttributeAsFloat(const char* AttrName) const;
-    /// return the named attribute as a double floating point (does type checking)
-    const char* getAttribute(const char* AttrName) const;
+
+    /// return the named attribute as an integer (does type checking); if missing return
+    /// defaultValue
+    long getAttributeAsInteger(const char* AttrName, const char* defaultValue = nullptr) const;

```

Comment:
And what is supposed to happen if this is used
`reader.getAttributeAsInteger("value", "test")` assuming value doesn't exist?

Reply:
You would get a 0 response. Assuming that an exception is the preferred response to a coding error, I've done #15521    Please advise if another approach is preferred.  I have not caught and rethrown the error with added details.

Reply:
Should be fine for now.